libraries/Microsoft.Bot.Connector/Authentication/EndorsementsValidator.cs

```diff
@@ -17,18 +17,18 @@ public static class EndorsementsValidator
         /// <see cref="Schema.Activity.ChannelId"/> property is set to "webchat" and the signing party
         /// of the JWT token must have a corresponding endorsement of “Webchat”.
         /// </summary>
-        /// <param name="channelId">The ID of the channel to validate, typically extracted from the activity's
+        /// <param name="expectedEndorsement">The expected endorsement. Generally the ID of the channel to validate, typically extracted from the activity's
         /// <see cref="Schema.Activity.ChannelId"/> property, that to which the Activity is affinitized.</param>
         /// <param name="endorsements">The JWT token’s signing party is permitted to send activities only for
         /// specific channels. That list, the set of channels the service can sign for, is called the the endorsement list.
         /// The activity’s <see cref="Schema.Activity.ChannelId"/> MUST be found in the endorsement list, or the incoming
         /// activity is not considered valid.</param>
         /// <returns>True if the channel ID is found in the endorsements list; otherwise, false.</returns>
-        public static bool Validate(string channelId, HashSet<string> endorsements)
+        public static bool Validate(string expectedEndorsement, HashSet<string> endorsements)
```

Comment:
Following the logic above, I expect that new endorsements are added to the channel ID endorsement.

We could have a mode where all endorsements are removed, but this is advanced and I'm not sure we have a use case for it.

Reply:
Ahh, I see how the channel ID endorsement verification is enforced at the layer above. Seems OK to me. Maybe just comment that multiple endorsements are enforced by calling this method multiple times.

Reply:
Reading more through the comment on this method, it should be rewritten to something more like:

        /// <summary>
        /// Verify that the specified endorsement exists on the JWT token. Call this method multiple times to validate multiple endorsements.
        /// </summary>
        /// <remarks>
        /// JWT token signing keys contain endorsements matching channel IDs they're approved to sign for. They also can contain keywords representing compliance certifications. This code ensures that a channel ID or compliance certification is present on the signing key used for the request's token.
        /// </remarks>

Reply:
Yes, EndorsementsValidator.Validate is only a collection existence check which was somewhat underwhelming to find out.